WooCommerce/Classes/POS/Presentation/Barcode Scanner Setup/PointOfSaleBarcodeScannerSetup.swift

```diff
@@ -114,3 +121,99 @@ private extension PointOfSaleBarcodeScannerSetup {
 #Preview {
     PointOfSaleBarcodeScannerSetup(isPresented: .constant(true))
 }
+
+/// A container view that animates changes in its child content with a fade-out and fade-in transition,
+/// while also smoothly animating changes in height.
+///
+/// - On first appear: content shows instantly (no fade).
+/// - On content change: fades out old content, replaces it, then fades in new content.
+/// - Handles height changes with a spring animation.
+///
+@available(iOS 17.0, *)
+private struct AnimatedTransitionContainer<Content: View, ID: Equatable>: View {
+    let maxWidth: CGFloat
+    let maxHeight: CGFloat
+    let contentID: ID
+    let contentBuilder: () -> Content
+
+    @State private var visibleContent: Content
+    @State private var previousID: ID
+    @State private var animatedHeight: CGFloat = 0
+    @State private var isVisible: Bool = true
+    @State private var hasAppeared: Bool = false
+
+    private let animationDuration: CGFloat = 0.3
+
+    init(
+        maxWidth: CGFloat,
+        maxHeight: CGFloat,
+        id: ID,
+        @ViewBuilder content: @escaping () -> Content
+    ) {
+        self.maxWidth = maxWidth
+        self.maxHeight = maxHeight
+        self.contentID = id
+        self.contentBuilder = content
+        self._visibleContent = State(initialValue: content())
+        self._previousID = State(initialValue: id)
+    }
+
+    var body: some View {
+        visibleContent
+            .opacity(isVisible ? 1 : 0)
+            // First layout pass: constrain content to let scrollView in content to configure itself
+            .frame(width: maxWidth)
+            .frame(maxHeight: maxHeight)
+            .fixedSize(horizontal: false, vertical: true)
+            // Measure the actual height after ScrollView has decided if it needs to scroll
+            .background(
+                GeometryReader { proxy in
+                    Color.clear
+                        .onAppear {
+                            updateSize(to: proxy.size.height)
+                        }
+                        .onChange(of: proxy.size) { newSize in
+                            updateSize(to: newSize.height)
+                        }
+                }
```

Comment:
Non-blocking: similar to the comment below, [`measureHeight` modifier](https://github.com/woocommerce/woocommerce-ios/blob/494ef48156e62f46bac4c539a3390d2e38557fb1/WooCommerce/Classes/View%20Modifiers/View%2BMeasurements.swift#L10) is available for animating the height in the callback.